polynote-spark/src/main/scala/polynote/kernel/SparkPolyKernel.scala

```diff
@@ -86,8 +86,8 @@ class SparkPolyKernel(
   // TODO: any better way to handle this?
   override protected lazy val symbolTable = realOutputPath.get.map {
     outputDir =>
-      val updatedClassLoader = GlobalInfo.genNotebookClassLoader(dependencies, globalInfo.classPath, outputDir, parentClassLoader)
-      new RuntimeSymbolTable(GlobalInfo(globalInfo.global, globalInfo.classPath, updatedClassLoader), statusUpdates)
+      val updatedClassLoader = KernelContext.genNotebookClassLoader(dependencies, kernelContext.classPath, outputDir, parentClassLoader)
+      new RuntimeSymbolTable(KernelContext(kernelContext.global, kernelContext.classPath, updatedClassLoader), statusUpdates)
```

Comment:
Hm, is it at all dangerous to be making new `KernelContext`s here? Will the globals still match now that there are multiple? I guess we've been doing it for a while and haven't run into any problems...

Reply:
I think it should generally be OK, as long as the old `KernelContext` doesn't get passed to anything. It's not ideal though (has to be done in this case, unless we make the `classLoader` mutable).